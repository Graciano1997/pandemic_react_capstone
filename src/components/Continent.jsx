import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';
import { getCountriesCases } from '../redux/country/countrySlice';
import style from '../assets/style/Continents.module.css';

function Continent({
  map, continent, countries, active, todayCases, todayRecovered, theme,
}) {
  const dispatch = useDispatch();
  const navegate = useNavigate();
  const handlerClick = async () => {
    const countriesList = countries.toString();
    await dispatch(getCountriesCases(countriesList));
    navegate('/countries');
  };

  return (
    <div className={style.continentItem} style={{ background: theme }}>
      <div className={style.continentImage}>
        <img src={map} alt="continent" className={style.continentImage} />
      </div>
      <div className={style.continentSituation}>
        <button type="button" onClick={handlerClick}>
          <FontAwesomeIcon icon={faCircleArrowRight} className={style.continentButton} />
        </button>
        <div className={style.continentStatus}>
          <h2>
            {continent}
          </h2>
          <h3>
            {`active: ${active}`}
          </h3>
          <h3>
            {`Today: ${todayCases}`}
          </h3>
          <h3>
            {`Today Recovered: ${todayRecovered}`}
          </h3>
        </div>
      </div>
    </div>
  );
}

Continent.propTypes = {
  continent: PropTypes.string.isRequired,
  map: PropTypes.string.isRequired,
  active: PropTypes.number.isRequired,
  theme: PropTypes.string.isRequired,
  todayCases: PropTypes.number.isRequired,
  todayRecovered: PropTypes.number.isRequired,
  countries: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Continent;
