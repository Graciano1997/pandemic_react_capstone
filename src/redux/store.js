import { configureStore } from '@reduxjs/toolkit';
import continentReducer from './continent/continentSlice';

const store = configureStore({
  reducer: {
    continent: continentReducer,
  },
});

export default store;
