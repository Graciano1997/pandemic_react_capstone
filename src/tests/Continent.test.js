import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Continent from '../components/Continent';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('Testing the Continents Rendering', () => {
  test('render the Africa Continents ', () => {
    const styleTheme = '#00000038';
    const standard = 'transparent';
    const standardContinents = ['Africa', 'Asia', 'Europe'];

    const continentsRendered = render(
      <Provider store={store}>
        <BrowserRouter>
          <Continent
            key={1}
            continent={'Africa'}
            population={5000000000000}
            cases={0}
            active={2000}
            todayCases={0}
            recovered={0}
            todayRecovered={20}
            map={'https://www.pngplay.com/wp-content/uploads/6/Africa-White-Map-Transparent-Background.png'}
            countries={['Angola', 'Cabo Verde']}
            theme={standardContinents.includes('Africa') ? styleTheme : standard}
          />
        </BrowserRouter>
      </Provider>,
    );
    expect(continentsRendered).toMatchSnapshot();
  });
  test('render the Africa Continents ', () => {
    const styleTheme = '#00000038';
    const standard = 'transparent';
    const standardContinents = ['Africa', 'Asia', 'Europe'];

    render(
      <Provider store={store}>
        <BrowserRouter>
          <Continent
            key={1}
            continent={'Africa'}
            population={5000000000000}
            cases={0}
            active={2000}
            todayCases={0}
            recovered={0}
            todayRecovered={20}
            map={'https://www.pngplay.com/wp-content/uploads/6/Africa-White-Map-Transparent-Background.png'}
            countries={['Angola', 'Cabo Verde']}
            theme={standardContinents.includes('Africa') ? styleTheme : standard}
          />
        </BrowserRouter>
      </Provider>,
    );
    const element = screen.getByText(/Africa/i);
    expect(element).toBeInTheDocument();
  });
});
