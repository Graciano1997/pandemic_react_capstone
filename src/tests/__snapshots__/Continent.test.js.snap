// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Continents Rendering render the Africa Continents  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="continentItem"
        style="background: rgba(0, 0, 0, 0.22);"
      >
        <div
          class="continentImage"
        >
          <img
            alt="continent"
            class="continentImage"
            src="https://www.pngplay.com/wp-content/uploads/6/Africa-White-Map-Transparent-Background.png"
          />
        </div>
        <div
          class="continentSituation"
        >
          <button
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-arrow-right continentButton"
              data-icon="circle-arrow-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="continentStatus"
          >
            <h3>
              Africa
            </h3>
            <h3>
              active: 2000
            </h3>
            <h3>
              Today: 0
            </h3>
            <h3>
              Today Recovered: 20
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="continentItem"
      style="background: rgba(0, 0, 0, 0.22);"
    >
      <div
        class="continentImage"
      >
        <img
          alt="continent"
          class="continentImage"
          src="https://www.pngplay.com/wp-content/uploads/6/Africa-White-Map-Transparent-Background.png"
        />
      </div>
      <div
        class="continentSituation"
      >
        <button
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right continentButton"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="continentStatus"
        >
          <h3>
            Africa
          </h3>
          <h3>
            active: 2000
          </h3>
          <h3>
            Today: 0
          </h3>
          <h3>
            Today Recovered: 20
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the Continents Rendering render the Continents Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="continentItem"
      >
        <div
          class="continentImage"
        >
          <img
            alt="continent"
            class="continentImage"
          />
        </div>
        <div
          class="continentSituation"
        >
          <button
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-arrow-right continentButton"
              data-icon="circle-arrow-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="continentStatus"
          >
            <h3 />
            <h3>
              active: undefined
            </h3>
            <h3>
              Today: undefined
            </h3>
            <h3>
              Today Recovered: undefined
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="continentItem"
    >
      <div
        class="continentImage"
      >
        <img
          alt="continent"
          class="continentImage"
        />
      </div>
      <div
        class="continentSituation"
      >
        <button
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-arrow-right continentButton"
            data-icon="circle-arrow-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256a256 256 0 1 0 512 0A256 256 0 1 0 0 256zM281 385c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l71-71L136 280c-13.3 0-24-10.7-24-24s10.7-24 24-24l182.1 0-71-71c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L393 239c9.4 9.4 9.4 24.6 0 33.9L281 385z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="continentStatus"
        >
          <h3 />
          <h3>
            active: undefined
          </h3>
          <h3>
            Today: undefined
          </h3>
          <h3>
            Today Recovered: undefined
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
